pf4j/src/main/java/org/pf4j/AbstractExtensionFinder.java

```diff
@@ -35,6 +37,8 @@
 
     protected PluginManager pluginManager;
     protected volatile Map<String, Set<String>> entries; // cache by pluginId
+    protected volatile Map<String, ExtensionInfo> extensionInfos; // cache extension infos by class name
+    protected Boolean checkForExtensionDependencies = null;
```

Comment:
Why not simple `boolean`? We need three state  (false, true, null) here?
I see that `null` check is made in https://github.com/pf4j/pf4j/pull/270/files#diff-1defd675a225b032b73ef2abcb518fd2R235 but I don't know if a simple `false` check is not enough.

Reply:
I understand that this flag is used to activate explicitly the extensions that require (optional) plugins feature. This flag is activated implicitly by PF4J if the plugin that contains the extension declares some optional plugins.
**Example** 
I have plugin `P1` that declares plugin `P2` as a optional dependency (via `MANIFEST.MF` or `plugin.properties`). In plugin `P1` I have an extension `E1` that requires optional plugin `P2`. To specify this thing, I will add `P2` in the `plugins` attribute of `E1` @Extension annotation.

Reply:
Basically the `checkForExtensionDependencies` has three possible states. If `checkForExtensionDependencies` is `null`, the automatic / implicit detection of optional plugin is executed.

But maybe the developer don't like to use this automatic detection at all. Therefore he might call `setCheckForExtensionDependencies` after he has created his `AbstractExtensionFinder` instance. In this case, `checkForExtensionDependencies` is either `true` or `false` and the automatic detection is disabled.

Or in other words: If `checkForExtensionDependencies` is `null`, it hasn't been set explicitly by the developer and the automatic / implicit detection of optional plugin is executed.

Reply:
In this case I think that I prefer an `enum` with three possible values, something like `NONE`, `IMPLICIT`, `EXPLICIT` (or better names). But it's my preference, nothing more, so I think that I can live with Boolean :)